app/src/main/java/org/sagebase/crf/CrfActivitiesFragment.java

```diff
@@ -84,9 +86,21 @@ public void taskSelected(SchedulesAndTasksModel.TaskScheduleModel task) {
         List<Object> tasks = new ArrayList<>();
 
         for (SchedulesAndTasksModel.ScheduleModel scheduleModel : model.schedules) {
-            tasks.addAll(scheduleModel.tasks);
+            for (SchedulesAndTasksModel.TaskScheduleModel task : scheduleModel.tasks) {
+                if (task.taskID != null && !hiddenActivityIdentifiers().contains(task.taskID)) {
+                    tasks.add(task);
+                }
+            }
         }
 
         return tasks;
     }
+
+    public List<String> hiddenActivityIdentifiers() {
+        String [] hideTheseActivities = new String [] {
+                CrfDataProvider.CLINIC1,
+                CrfDataProvider.CLINIC2};
+
+        return new ArrayList<>(Arrays.asList(hideTheseActivities));
+    }
```

Comment:
This method recreates the array every time it's called (which is for every task for every schedule every time we load the activities page).

It would be better to have a static final List as a top-level class member. You can use ImmutableList so you don't have to do wonky things with String arrays. Also, consider a Set instead of a List.

`public static final Set&lt;String&gt; HIDDEN_ACTIVITY_IDENTIFIERS = ImmutableSet.of(CrfDataProvider.CLINIC1, CrfDataProvider.CLINIC2);`

Reply:
Nice, I didn't know about ImmutableSet.of, good tip.